packages/webapp/src/hooks/useOfflineCheck.ts

```diff
@@ -0,0 +1,73 @@
+import { useState, useEffect } from "react";
+import { toast } from "sonner";
+
+async function performNetworkCheck() {
+  const endpoint = "https://www.google.com";
+  try {
+    const response = await fetch(endpoint, {
+      method: "HEAD",
+      cache: "no-cache",
+      mode: "no-cors",
+    });
+    return response.ok;
+  } catch (error) {
+    throw error; // Rethrow for retries
+  }
+}
+
+function useOfflineCheck(options: any = {}) {
+  const {
+    // Optional configuration properties:
+    threshold = 1000, // Minimum interval between offline checks (ms)
+    maxRetries = 3, // Maximum number of retries before assuming offline
+    retryDelay = 1500, // Delay between retries (ms)
+  } = options;
+
+  const [isOffline, setIsOffline] = useState(false);
+  const [retries, setRetries] = useState(0);
```

Comment:
These two being React state is not needed I think, as it means the `useOfflineCheck` we will rerender each time we increment the amount of retries or set `isOffline` — this is unnecessary, the `toast.error` function will trigger a rerender itself and nothing else is needed. You can make those `let` variables and mutate them directly.

Reply:
i think i was overthinking this one and i have brought forward logic i was using before you introduced the sonner component. we don't need the isOffline state again.